src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -707,8 +709,8 @@ export class ResultsViewPageStore {
             this.samples
         ],
         invoke: () => {
-            if (this.genes.result) {
-                return Promise.resolve(_.reduce(_.uniq(this.genes.result.map(g => g.chromosome)), (map: { [chromosome: string]: MobxPromise<CopyNumberSeg[]> }, chromosome: string) => {
+            if (this.refGenes.result) {
+                return Promise.resolve(_.reduce(_.uniq(this.refGenes.result.map(g => g.chromosome)), (map: { [chromosome: string]: MobxPromise<CopyNumberSeg[]> }, chromosome: string) => {
```

Comment:
this code is kind of hard to read.  i would first make a variable uniqueChromosommes and assign map to it.  Then call reduce on variable.

Reply:
good idea! will make changes.

Reply:
made a variable **_uniqueReferenceGenes_** and call reduce on the variable.